### PR TITLE
30 url origin validation

### LINKED ISSUE #30: Error handling: Web Address must contain "https://" or "http://"
The Web Address must contain a scheme (e.g., "https://") and a host (domain + port), and must not include a path.

If the URI is not well-formed, an error is displayed suggesting the user include "https://".

If the URI is well-formed but includes a path, the path can be stripped.

## Implementation Details

* Start using web URL class for validation.
* Extra testing to be sure URL class is sufficient.

## Visual Design
**InVision**
Helper: https://mozilla.invisionapp.com/share/ENOL84959AK#/339614514_Logins_-_New_Entry_-_Address_Focused_-Light-
Error: https://mozilla.invisionapp.com/share/ENOL84959AK#/339614519_Logins_-_New_Entry_-_Error_-_Address_-Light-

**Zeplin**
Helper: https://zpl.io/andLDWA
Error: https://zpl.io/aNKzXWk

### PR DESCRIPTION
**Pull request by meandavejustice**
Fixes #30 

## Screenshots

error-new-input-lockbox
focus-new-inputs-lockbox

### CODE REVIEW

File: src/list/components/item-fields.js [hunk 1]
```diff
@@ -121,18 +124,36 @@ export class EditItemFields extends React.Component {
...
-        </label>
+        <div className={styles.inputWrap}>
+          <label>
+            <Localized id="item-fields-origin">
+              <LabelText>oRIGIn</LabelText>
+            </Localized>
+            <Localized id="item-fields-origin-input" attrs={{placeholder: true}}>
+              <Input className={styles.originInput} type="text"
+                     disabled={!!fields.itemId}
+                     pattern={originPattern}
```
Comment (6a68) [inline src/list/components/item-fields.js]: If you want to just use html5 input validation, I'd use `type=url` here, and skip the `pattern`. That should get us everything we need. You could add a prop like `validationAttrs` to pass `type` or `pattern` or both down as needed.

File: src/locales/en-US/list.ftl [hunk 2]
```diff
@@ -20,6 +20,8 @@ item-fields-title-input =
 item-fields-origin = Website Address
 item-fields-origin-input =
   .placeholder = https://www.example.com
+item-fields-origin-error-message = Must include “https://“ or “http://“
```
[the remark this answers is not in the document]
Comment (linuxwolf) [reply src/locales/en-US/list.ftl]: > @linuxwolf could we say something like "Must include 'https://' or 'http://' or another valid URL scheme"?

I'll note this comes from the UX prototypes, and I believe this text in particular has gone through a few revisions.

Consider the target user, which generally have little-to-no concept what a URL is, let alone the scheme portion.  This is an attempt to balance that; let the user put whatever valid URL, and point them to the success path for >90% of people.

[the remark this answers is not in the document]
Comment (meandavejustice) [reply src/locales/en-US/list.ftl]: Not sure what we want to do here. I addressed the other comments and tests are now running again.

[the remark this answers is not in the document]
Comment (linuxwolf) [reply src/locales/en-US/list.ftl]: apologies for the delay; I think the current text is best for now.  We can adjust it later based on user feedback (with the understood caveat that it'll require l10n work).  We are trying to strike a balance between people that pretty much know what they are doing, and people that just want to get some login to be added already.

**Comment by 6a68** [comment]
@meandavejustice are you waiting on anything here, or is this good to land?

**Comment by meandavejustice** [comment]
@6a68 Was waiting on https://github.com/mozilla-lockbox/lockbox-addon/pull/115#discussion_r270966058. Its good to merge now

**Pull request merged**